### PR TITLE
DOC: Updated notes and examples for np.insert.

### LINKED ISSUE #25895: DOC: update the note for "insert" so that it points to the explanation of why `...0...` and `...[0]...` are wildly different.
### Issue with current documentation:

https://numpy.org/doc/stable/reference/generated/numpy.insert.html#numpy.insert currently has a note that reads:

>Note that for higher dimensional inserts obj=0 behaves very different from obj=[0] just like arr[:,0,:] = values is different from arr[:,[0],:] = values.

Which doesn't let readers find out what the specific difference in behaviour is, or why that difference exists.

### Idea or request for content:

Link out to whatever page(s) explain this difference in behaviour, so that people aren't left with more questions than they had before they saw that note =)

### PR DESCRIPTION
**Pull request by jud-sdev**
This closes issue #25895. Continues PR #26114.
I added a note with a link.
I changed the examples to use non-repeating numbers, 
to make the output more obvious.
I added an example to show the difference between 
`obj=1` and `obj=[1]`.

[skip actions] [skip azp] [skip cirrus]

### CODE REVIEW

**Comment by mattip** [comment]
Continuation of #26114

File: numpy/lib/_function_base_impl.py [hunk 1]
```diff
@@ -5418,43 +5418,49 @@ def insert(arr, obj, values, axis=None):
...
+    showing how ``obj=[1]`` behaves different from ``obj=1``:
+
+    >>> np.insert(a, [1], [[7],[8],[9]], axis=1)
+    array([[0, 7, 1],
+           [2, 8, 3],
+           [4, 9, 5]])
+    >>> np.insert(a, 1, [[7],[8],[9]], axis=1)
+    array([[0, 7, 8, 9, 1],
+           [2, 7, 8, 9, 3],
+           [4, 7, 8, 9, 5]])
```
Comment (bmwoodruff) [inline numpy/lib/_function_base_impl.py:5448]: This line was added to help address the issue that led to this PR.  We talked about putting a "See the examples below" comment in the Notes section, but then the Notes and Examples sections are not independent (which could lead to later problems). I think this is enough to help a user quickly see the difference between using `obj=1` and `obj=[1]`.

File: numpy/lib/_function_base_impl.py [hunk 2]
```diff
@@ -5418,43 +5418,49 @@ def insert(arr, obj, values, axis=None):
     -----
     Note that for higher dimensional inserts ``obj=0`` behaves very different
     from ``obj=[0]`` just like ``arr[:,0,:] = values`` is different from
-    ``arr[:,[0],:] = values``.
+    ``arr[:,[0],:] = values``. This is because of the difference between basic
+    and advanced :ref:`indexing <basics.indexing>`.
```
Comment (bmwoodruff) [inline numpy/lib/_function_base_impl.py:5422]: I think this appropriately points the user to a reference if they aren't sure why things behave differently. It's not a one line fix, but the example added below hopefully addresses that.

File: numpy/lib/_function_base_impl.py [hunk 3]
```diff
@@ -5418,43 +5418,49 @@ def insert(arr, obj, values, axis=None):
...
 
     Examples
     --------
-    >>> a = np.array([[1, 1], [2, 2], [3, 3]])
+    >>> a = np.arange(6).reshape(3, 2)
```
Comment (bmwoodruff) [inline numpy/lib/_function_base_impl.py:5426]: Swapping to a different example which has no repeated numbers hopefully helps make it obvious which things are changing. The rest of the changes below are to make sure newly inserted objects are different.

**Review by bmwoodruff — APPROVED** [review]
Looks good to me.

**Comment by bmwoodruff** [comment]
@charris , thanks for mentioning this one today at the community meeting. I think it's cleaned up to a point where it might address the issues discussed last week.

**Comment by mattip** [comment]
Thanks @jud-sdev, @bmwoodruff

**Pull request merged**